src/core/dbus-manager.c

```diff
@@ -1968,13 +1969,16 @@ static int method_enable_unit_files_generic(
         if (r < 0)
                 return r;
 
+        flags = (runtime & UNIT_FILE_RUNTIME) |
+                (force & UNIT_FILE_FORCE);
```

Comment:
this doesn't look right. runtime/force are a boolean, stored in an int, i.e. either 0 or != 0. thus you need something like:

```c
flags = (runtime ? UNIT_FILE_RUNTIME : 0) |
        (force ? UNIT_FILE_FORCE : 0);
```

You could also write it as:

```c
flags = (!!runtime * UNIT_FILE_RUNTIME) |
        (!!force * UNIT_FILE_FORCE)
```

But I must say I generally find the ternary syntax more obvious...